src/BenchmarkDotNet/Detectors/Cpu/Windows/PowershellWmiCpuDetector.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using BenchmarkDotNet.Helpers;
+using Perfolizer.Models;
+
+namespace BenchmarkDotNet.Detectors.Cpu.Windows;
+
+/// <summary>
+/// CPU information from output of the `wmic cpu get Name, NumberOfCores, NumberOfLogicalProcessors /Format:List` command.
+/// Windows only.
+/// </summary>
+internal class PowershellWmiCpuDetector : ICpuDetector
+{
+    private readonly string windowsPowershellPath =
+        $"{Environment.SystemDirectory}{Path.DirectorySeparatorChar}WindowsPowerShell{Path.DirectorySeparatorChar}" +
+        $"v1.0{Path.DirectorySeparatorChar}powershell.exe";
+
+    public bool IsApplicable() => OsDetector.IsWindows();
+
+    public CpuInfo? Detect()
+    {
+        if (!IsApplicable()) return null;
+
+        string programFiles = Environment.Is64BitOperatingSystem ? Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles) : Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86);
+
+        string powershell7PlusPath = $"{programFiles}{Path.DirectorySeparatorChar}Powershell{Path.DirectorySeparatorChar}";
+
+        bool checkForPowershell7Plus = true;
+
+        if (Directory.Exists(powershell7PlusPath))
+        {
+            //Use .Last so that we get the newest major PowerShell version
+           string[] subDirectories = Directory.EnumerateDirectories(powershell7PlusPath, "*", SearchOption.AllDirectories)
+               .ToArray();
+
+           string? subDirectory = null;
+
+           if (subDirectories.Any())
+           {
+               subDirectory = subDirectories.Where(x => Regex.IsMatch(x, "[0-9]"))
+                   .Select(numStr => new { Value = numStr, Number = int.Parse(numStr) })
+                   .OrderByDescending(x => x.Number)
+                   .Select(x => x.Value).FirstOrDefault();
+           }
+
+           if (subDirectory is not null)
+           {
+               powershell7PlusPath = $"{subDirectory}{Path.DirectorySeparatorChar}pwsh.exe";
+           }
+           else
+           {
+               checkForPowershell7Plus = false;
+           }
+        }
+
+        const string argList = $"{WmicCpuInfoKeyNames.Name}, " +
+                               $"{WmicCpuInfoKeyNames.NumberOfCores}, " +
+                               $"{WmicCpuInfoKeyNames.NumberOfLogicalProcessors}, " +
+                               $"{WmicCpuInfoKeyNames.MaxClockSpeed}";
+
+        string powershellPath;
+
+        // Optimistically, use Cross-platform new PowerShell when available but fallback to Windows PowerShell if not available.
+        if (checkForPowershell7Plus)
+        {
+            powershellPath = File.Exists(powershell7PlusPath) ? powershell7PlusPath : windowsPowershellPath;
+        }
+        else
+        {
+            powershellPath = windowsPowershellPath;
+        }
+
+        if (File.Exists(powershellPath) == false)
```

Comment:
Knowing the exact path to PowerShell can be beneficial, but we shouldn't assume that we can reliably determine it for all existing and future versions of Windows. If PowerShell isn't found in any of the known locations, we should fall back to invoking it simply as `"PowerShell"`, relying on the system to resolve it via the `PATH` environment variable. This provides a reasonable fallback mechanism without hardcoding assumptions that may not hold across different environments.